### PR TITLE
MAKE-1097 MAKE-1107 Improve logging and operationalize repobuilder code

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: repobuilder/job.go [hunk 1]
```diff
@@ -129,13 +141,28 @@ func (j *Job) linkPackages(dest string) error {
 		mirror := filepath.Join(dest, filepath.Base(pkg))
 		if j.release.IsDevelopmentBuild() {
 			if _, err := os.Stat(mirror); os.IsNotExist(err) {
-				err = errors.Wrap(os.Remove(mirror), "problem removing previous development build")
-				grip.Notice(err)
+				grip.Warning(message.WrapError(os.Remove(mirror), message.Fields{
```
Comment (Kimchelly) [inline repobuilder/job.go]: Why does this attempt to remove the directory if the directory already doesn't exist?

File: repobuilder/job.go [hunk 2]
```diff
@@ -228,54 +261,97 @@ func (j *Job) signFile(fileName, archiveExtension string, overwrite bool) error
...
 	if overwrite {
-		grip.Noticef("overwriting existing contents of file '%s' while signing it", fileName)
 		args = append(args, "--package-file-suffix", "")
 	} else {
 		// if we're not overwriting the unsigned source file
 		// with the signed file, then we should remove the
 		// signed artifact before. Unclear if this is needed,
 		// the cronjob did this.
-		grip.Warning(os.Remove(fileName + "." + archiveExtension))
+		grip.Warning(message.WrapError(os.Remove(fileName+"."+archiveExtension),
```
Comment (Kimchelly) [inline repobuilder/job.go]: Is this still fine if archiveExtension is empty?

File: repobuilder/job.go (hunk 1 above)
Comment (tychoish) [reply repobuilder/job.go]: it's always been this way. Will delete

File: repobuilder/job.go (hunk 2 above)
Comment (tychoish) [reply repobuilder/job.go]: I don't think this logic has changed and it doesn't lead to any problems, I'm not inclined to change it at this point.

**Pull request merged**